src/garage/torch/algos/vpg.py

```diff
@@ -0,0 +1,231 @@
+            conjunction with center_adv the advantages will be
+            standardized before shifting.
+        optimizer (object): The optimizer of the algorithm. Should be the
+            optimizers in torch.optim.
+        policy_ent_coeff (float): The coefficient of the policy entropy.
+            Setting it to zero would mean no entropy regularization.
+        use_softplus_entropy (bool): Whether to estimate the softmax
+            distribution of the entropy to prevent the entropy from being
+            negative.
+        stop_entropy_gradient (bool): Whether to stop the entropy gradient.
+        entropy_method (str): A string from: 'max', 'regularized',
+            'no_entropy'. The type of entropy method to use. 'max' adds the
+            dense entropy to the reward for each time step. 'regularized' adds
+            the mean entropy to the surrogate objective. See
+            https://arxiv.org/abs/1805.00909 for more details.
+    """
+
+    def __init__(
+            self,
+            env_spec,
+            policy,
+            baseline,
+            max_path_length=500,
+            policy_lr=1e-2,
+            n_samples=1,
+            discount=0.99,
+            gae_lambda=1,
+            center_adv=True,
+            positive_adv=False,
+            max_kl_step=0.01,
+            optimizer=None,
+            policy_ent_coeff=0.0,
+            use_softplus_entropy=False,
+            stop_entropy_gradient=False,
+            entropy_method='no_entropy',
+    ):
+        self.env_spec = env_spec
+        self.policy_lr = policy_lr
+        self.gae_lambda = gae_lambda
+        self.center_adv = center_adv
+        self.positive_adv = positive_adv
+        self.max_kl_step = max_kl_step
+        self.policy_ent_coeff = policy_ent_coeff
+        self.use_softplus_entropy = use_softplus_entropy
+        self.stop_entropy_gradient = stop_entropy_gradient
+        self.entropy_method = entropy_method
+        self.eps = 1e-8
+
+        self.maximum_entropy = (entropy_method == 'max')
+        self.entropy_regularzied = (entropy_method == 'regularized')
+        self._check_entropy_configuration(entropy_method, center_adv,
+                                          stop_entropy_gradient,
+                                          policy_ent_coeff)
+        self._episode_reward_mean = collections.deque(maxlen=100)
+        self.optimizer = optimizer(policy.get_module().parameters(),
+                                   lr=policy_lr)
+
+        filter = torch.full((1, 1, 1, max_path_length), discount * gae_lambda)
+        self.filter = torch.cumprod(filter, dim=-1)
+
+        super().__init__(policy=policy,
+                         baseline=baseline,
+                         discount=discount,
+                         max_path_length=max_path_length,
+                         n_samples=n_samples)
+
+    def _check_entropy_configuration(self, entropy_method, center_adv,
```

Comment:
Do we need this in a vpg implementation? When might someone to use an entropy gradient with vpg?

Reply:
@krzentner

Reply:
Maximum entropy RL is useful (and meaningful) even with VPG, right? That requires the entropy gradient.